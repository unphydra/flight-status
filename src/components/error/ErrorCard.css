.error-card-paper {
  margin: 20vh auto;
  max-width: 40vw;
  min-height: 40vh;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.error-card-typography {
  margin-bottom: 16px !important;
}
