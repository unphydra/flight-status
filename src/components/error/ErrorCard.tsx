import { Container, Paper, Typography } from "@mui/material";
import { ComponentType } from "react";

interface Props {
  title: string;
  description: string;
  ActionComponent?: ComponentType;
}

export default function ErrorCard({
  title,
  description,
  ActionComponent,
}: Props) {
  return (
    <Paper
      elevation={3}
      sx={{
        m: "20vh auto",
        maxWidth: "40vw",
        minHeight: "40vh",
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        textAlign: "center",
      }}
    >
      <Container>
        <Typography variant={"h5"} mb={2}>
          {title}
        </Typography>
        <Typography mb={2}>{description}</Typography>
        {ActionComponent && <ActionComponent />}
      </Container>
    </Paper>
  );
}
