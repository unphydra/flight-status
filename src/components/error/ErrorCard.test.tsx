import { render, screen } from "@testing-library/react";
import ErrorCard from "./ErrorCard";

describe("ErrorCard", () => {
  const title = "Error Title";
  const description = "Error Description";

  it("Should renders without crashing", () => {
    const { container } = render(
      <ErrorCard title={title} description={description} />
    );
    expect(container).toMatchSnapshot();
  });

  it("Should renders title and description", () => {
    render(<ErrorCard title={title} description={description} />);

    const titleElement = screen.getByText(title);
    const descriptionElement = screen.getByText(description);

    expect(titleElement).toBeInTheDocument();
    expect(descriptionElement).toBeInTheDocument();
  });

  it("renders action component if provided", () => {
    const ActionComponent = () => <button>Retry</button>;

    render(
      <ErrorCard
        title={title}
        description={description}
        ActionComponent={ActionComponent}
      />
    );

    const actionComponent = screen.getByRole("button", { name: "Retry" });

    expect(actionComponent).toBeInTheDocument();
  });
});
