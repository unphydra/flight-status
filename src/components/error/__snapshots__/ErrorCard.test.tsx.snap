// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorCard > Should renders without crashing 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-83nk81-MuiPaper-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ch2k2f-MuiTypography-root"
      >
        Error Title
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7caa22-MuiTypography-root"
      >
        Error Description
      </p>
    </div>
  </div>
</div>
`;
